### PR TITLE
feat(code-block): added code block component

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/3907

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
Looks good to me @mcoker . You @lwigh?

File: src/patternfly/components/CodeBlock/code-block-code.hbs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<code class="pf-c-code-block__code{{#if code-block-code--modifier}} {{code-block-code--modifier}}{{/if}}"
+  {{#if code-block-code--attribute}}
+    {{{code-block-code--attribute}}}
+  {{/if}}>
+  {{~>@partial-block~}}
```
Comment (mattnolting) [inline src/patternfly/components/CodeBlock/code-block-code.hbs:5]: Did you mean to add `~`?

File: src/patternfly/components/CodeBlock/code-block.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  --pf-c-code-block__content--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-code-block__pre--FontFamily: var(--pf-global--FontFamily--monospace);
+  --pf-c-code-block__pre--FontSize: var(--pf-global--FontSize--sm);
+
+  background-color: var(--pf-c-code-block--BackgroundColor);
+}
+
+.pf-c-code-block__header {
+  display: flex;
+  width: 100%;
```
Comment (mattnolting) [inline src/patternfly/components/CodeBlock/code-block.scss:17]: Don't think you need this.
```suggestion
```

**Review by mattnolting — COMMENTED** [review]
This looks awesome, great work! Just need to add `white-space` controls, and it's GTG!

Screen Shot 2021-03-23 at 2 09 17 PM

**Review by mattnolting — APPROVED** [review]
LPTM! 🎉

**Comment by lwrigh** [comment]
The demo looks great to me!

**Comment by mcarrano** [comment]
This looks good @mcoker . My only question is whether the gap between the bottom of the text and the expansion widget needs to be so wide. Is that just a function of the block contents or is that built-in spacing?

**Comment by mcoker** [comment]
@mcarrano That's an artifact of whitespace management in our examples, @redallen is working on a fix for us. This is what the expand button will look like when we can better manage the whitespace.

Screen Shot 2021-03-24 at 2 28 16 PM
Screen Shot 2021-03-24 at 2 30 24 PM

**Review by mcarrano — APPROVED** [review]
Thanks for the clarification @mcoker . I approve then!

**Comment by sahil143** [comment]
There should not be a need to put <br /> tag on each line inside <pre/>. It should handle the formatting itself.

**Comment by christianvogt** [comment]
@mcoker I agree with @sahil here. If we're displaying a code block, I don't want to have to replace new lines with a break tag.

**Comment by redallen** [comment]
The reason that the `<br>` tags are there is a workspace bug. #3956 should fix it.

**Pull request merged**